lib/spack/spack/compilers/apple_clang.py

```diff
@@ -41,46 +41,46 @@ def extract_version_from_output(cls, output):
     @property
     def cxx11_flag(self):
         # Spack's AppleClang detection only valid from Xcode >= 4.6
-        if self.real_version < spack.version.ver("4.0"):
+        if self.real_version < spack.version.ver("=4.0"):
```

Comment:
Yeah, should be, but I think comparison between range and exact version is more for internal use (just s.t. versionlist can use insertion sort). I think it may be suprising/confusing to people that  `1.5 > 1.0:2.0` is true.

Reply:
so wouldn't it be less of a footgun if VersionList's sort used some internal comparator, and the user-exposed operators compared based on the ends of the range?

Reply:
In principle it could use an internal comparator, but I think `<` should then just error for mixed Version / ClosedOpenRange since there's no natural total order (if you want 1 < 2:3 and 2:3 < 4, how do 2.5 and 2:3 compare)